Info panel

arscan opened the pull request:
~~Work in progress, do not merge yet~~.  Ready to merge.

I'm experimenting with where we can show/edit Title and Remarks for modules.  One option is to put a panel on the left side, which then gives us room to show some other things (like attributes set by this module, a list of states, etc).  The downside is that it takes away room for the chart.

Stable, but at the very least I need to style this before merging.  I'm not completely sold on having a panel on the left side either.  Putting up this PR for feedback.

arscan commented:
![screen shot 2018-01-17 at 8 48 32 am](https://user-images.githubusercontent.com/412901/35046079-f476a59c-fb63-11e7-9d2f-bf43ce5f9127.png)

Current status.  Needs styling at a minimum.

jawalonoski commented:
I don't mind the panel on the left.

You'll need to put the "palette of risk factors" somewhere anyway...

arscan commented:
![screen shot 2018-01-19 at 2 29 39 pm](https://user-images.githubusercontent.com/412901/35167959-4aef1b30-fd25-11e7-9205-8be22fdf4515.png)

@schreiaj and I decided to go with one panel for now, and put the state editor in the bottom 2/3rd of the left panel.

Highlights of changes
* Shows module name, remarks, which are editable
* state editor in left panel below module name/remarks
* initial graph position should be much smarter now... it will zoom and fit to available space on load
* changed selected color to blue... honestly I think it is a little hard to read but I agree that red is not great either
* when zoomed way out, the lines were too small.  i have some rough logic in there to make them bolder, but it needs some tweaking still
* fixed the save modal (it never updated when switching modules, etc)
* renamed save modal to download, and renamed the class (was called 'code' before, which is the same name as our code set component)
* I axed the Menu component and just put the nav bar in the top.  Still need to refactor modals into something more react-friendly, but that isn't a high priority
* probably some other stuff

I didn't style the state editor... figured I'd wait until that was complete before looking at it.

Could you play around with this a bit before merging @schreiaj?  I'm a little concerned because I had to merge in a lot of changes from master, which hopefully didn't get lost.  Also, there is still a bug when you click on 'Wellness_Encounter', but that's an existing issue.
